source/Octopus.Tentacle.Client/Retries/RpcCallRetryHandler.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Polly;
+using Polly.Timeout;
+
+namespace Octopus.Tentacle.Client.Retries
+{
+    internal class RpcCallRetryHandler
+    {
+        readonly TimeoutStrategy timeoutStrategy;
+
+        public RpcCallRetryHandler(TimeSpan retryTimeout, TimeoutStrategy timeoutStrategy)
+        {
+            this.timeoutStrategy = timeoutStrategy;
+            RetryTimeout = retryTimeout;
+        }
+
+        public TimeSpan RetryTimeout { get; }
+
+        public async Task<T> ExecuteWithRetries<T>(
+            Func<CancellationToken, Task<T>> action,
+            Func<Exception, TimeSpan, int, TimeSpan, CancellationToken, Task>? onRetryAction,
+            Func<TimeSpan, CancellationToken, Task>? onTimeoutAction,
```

Comment:
@nathanwoctopusdeploy it isn't super obvious what each of the params to the Funcs are. E.g. what is the timespan?

Could these be delegates so that the parameters can be named and or documented.